test/unit/signup/account_manager_test.rb

```diff
@@ -126,23 +126,29 @@ class ProviderAccountManagerTest < Signup::AccountManagerTest
       assert_equal 'API', account.first_service!.name
     end
 
-    test 'creating a provider will create default Backend if api_as_product is disabled' do
+    test 'creating a provider will create default Backend regardless of api_as_product' do
       Account.any_instance.stubs(:provider_can_use?).returns(false)
       Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(false)
       account = signup_account_manager.create(signup_params).account
-      assert_equal BackendApi.default_api_backend, account.default_service.api_backend
+      assert_expected_backend_api(account)
     end
 
     test 'creating a provider will create default Backend if api_as_product is enabled' do
       Account.any_instance.stubs(:provider_can_use?).returns(false)
       Account.any_instance.stubs(:provider_can_use?).with(:api_as_product).returns(true)
       account = signup_account_manager.create(signup_params).account
-      assert account.backend_apis.first
-      assert_equal BackendApi.default_api_backend, account.default_service.api_backend
+      assert_expected_backend_api(account)
```

Comment:
```suggestion
      assert (backend_api = account.backend_apis.first)
      assert_equal BackendApi.default_api_backend, backend_api.private_endpoint
```

Reply:
This is exactly what we had and I am trying to tackle this:
https://github.com/3scale/porta/pull/1292#discussion_r330740443

Reply:
https://github.com/3scale/porta/pull/1294#discussion_r330980774

Reply:
It is not the same. I changed second assertion.

@hallelujah's suggestion is valid. I agree. To tackle that, let's add:
```
assert_equal 1, account.backend_apis.count
FactoryBot.create(:simple_service, account: account)
assert_equal 1, account.backend_apis.count
```

Reply:
I disagree with that last way to test it because creating a new service by the factory is not testing the ProviderAccountManager what is what this test should be for